plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,503 @@
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(mastersCmd)
+
+	masters, mastersErr := mastersCmd.Result()
+	if mastersErr != nil {
+		acc.AddError(mastersErr)
+		return
+	}
+
+	for _, master := range masters {
+		master, masterOk := master.([]interface{})
+		if !masterOk {
+			continue
+		}
+
+		m := toMap(master)
+
+		masterName, masterNameOk := m["name"]
+		if !masterNameOk {
+			acc.AddError(fmt.Errorf("unable to resolve master name"))
+			return
+		}
+
+		quorumCmd := redis.NewStringCmd("sentinel", "ckquorum", masterName)
+		// We check the command result below
+		//nolint:errcheck
+		client.sentinel.Process(quorumCmd)
+
+		_, quorumErr := quorumCmd.Result()
+
+		sentinelMastersTags, sentinelMastersFields, err := convertSentinelMastersOutput(client.baseTags(), m, quorumErr)
+		if err == nil {
+			acc.AddFields(measurementMasters, sentinelMastersFields, sentinelMastersTags)
+		} else {
+			acc.AddError(err)
+		}
+
+		gatherReplicaStats(acc, client, masterName)
+		gatherSentinelStats(acc, client, masterName)
+	}
+}
+
+func gatherReplicaStats(
+	acc telegraf.Accumulator,
+	client *RedisSentinelClient,
+	masterName string,
+) {
+	replicasCmd := redis.NewSliceCmd("sentinel", "replicas", masterName)
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(replicasCmd)
+
+	replicas, replicasErr := replicasCmd.Result()
+	if replicasErr != nil {
+		acc.AddError(replicasErr)
+		return
+	}
+
+	for _, replica := range replicas {
+		if replica, replicaOk := replica.([]interface{}); replicaOk {
+			rm := toMap(replica)
+
+			replicaTags, replicaFields, err := convertSentinelReplicaOutput(client.baseTags(), masterName, rm)
+			if err == nil {
+				acc.AddFields(measurementReplicas, replicaFields, replicaTags)
+			} else {
+				acc.AddError(err)
+			}
+		}
+	}
+}
+
+func gatherSentinelStats(
+	acc telegraf.Accumulator,
+	client *RedisSentinelClient,
+	masterName string,
+) {
+	sentinelsCmd := redis.NewSliceCmd("sentinel", "sentinels", masterName)
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(sentinelsCmd)
+
+	sentinels, sentinelsErr := sentinelsCmd.Result()
+	if sentinelsErr != nil {
+		acc.AddError(sentinelsErr)
+		return
+	}
+
+	for _, sentinel := range sentinels {
+		if sentinel, sentinelOk := sentinel.([]interface{}); sentinelOk {
+			sm := toMap(sentinel)
+
+			sentinelTags, sentinelFields, err := convertSentinelSentinelsOutput(client.baseTags(), masterName, sm)
+			if err == nil {
+				acc.AddFields(measurementSentinels, sentinelFields, sentinelTags)
+			} else {
+				acc.AddError(err)
+			}
+		}
+	}
+}
+
+// converts `sentinel masters <name>` output to tags and fields
+func convertSentinelMastersOutput(
+	globalTags map[string]string,
+	master map[string]string,
+	quorumErr error,
+) (map[string]string, map[string]interface{}, error) {
+	tags := make(map[string]string)
+	for k, v := range globalTags {
+		tags[k] = v
+	}
```

Comment:
I think
```suggestion
	tags := globalTags
```
if I'm not totally mistaken...